### ISSUE TITLE
Does not work with App Router routes having parameters (slugs)

### ISSUE THREAD

**Issue by salilponde** [davidnx/NextjsStaticHosting-AspNetCore#16]
I am using App Router in Next 13.4. I have a route /app/containers/[containerId]/logs/page.js. 

When I open http://whatever_url_and_port/containers/ab112233/logs, ASP.NET Core returns 404 not found response. There is no HTML file for the logs page in the static output generated by NextJS. I can only see a JS file in the chunks directory.

It appears that the package looks for HTML files to serve and there is no HTML file generated for above page so we get 404 response. Is there any way this can be supported? Or is there a workaround?

**Comment by salilponde** [comment]
I have converted by app to use Pages Router instead of App Router as a workaround and it works fine now.

**Comment by davidnx** [comment]
@salilponde it's a Next.js issue (err feature?). Long story short, for now it seems best to continue to use Pages Router like you did.

Digging in a little deeper, there are a few things worth mentioning:

1. Next.js App Router has limitations when it comes to exporting apps with dynamic routes (for understandable reasons)
2. Next.js has improved detection of this scenario in recent versions. At least on `14.1.0`, if you attempt to build when you have an App Router page with a dynamic segment. For example:

   ```tsx
   // /src/app/tasks/[taskId]/page.tsx:
   export default function TaskPage({ params }: { params: { taskId: string } }) {
     const { taskId } = params;
     return (
       <div>
         Some task: {taskId}
       </div>
     );
   }
   ```

   Attempting to build produces this error:
   ```
   Error: Page "/tasks/[taskId]" is missing "generateStaticParams()" so it cannot be used with "output: export" config.
   ```
3. If you add `generateStaticParams` as the error suggests, you can have Next.js pre-render specific variants of your page for those specific values of the dynamic segment, but that doesn't give you dynamic segments like you wanted:
   ```tsx
   export function generateStaticParams() {
     return [{ taskId: 'task1' }]
   }
 
   export default function TaskPage({ params }: { params: { taskId: string } }) {
     const { taskId } = params;
     return (
       <div>
         Some task: {taskId}
       </div>
     );
   }
   ```
4. I tried "fooling" Next.js into doing what we want by having `generateStaticParams` return a single entry with the placeholder name as the value. This didn't work and seems to indicate a bug in Next.js -- no output was produced in this case!
   ```tsx
   export function generateStaticParams() {
     return [{ taskId: '[taskId]' }];
   }
   ```
5. Another attempt to "fool" Next.js worked slightly better: This time I set the placeholder value also to something in brackets, just not precisely the current segment name. This time I **did** get a `out/tasks/[foo].html` and this MAY be a viable path forward for you. **However**, I did not go so far as to verify if you would be able to apply state correctly once the page hydrates on the client. The little I did find is that, in the statically generated page, the param provided to the page component had value `%5Bfoo%5D` (i.e. the URL escaped version of `[foo]`). This feels very hacky and goes beyond what I would be comfortable relying on for anything of importance.
   ```tsx
   export function generateStaticParams() {
     return [{ taskId: '[foo]' }];
   }
   ```

In conclusion, unfortunately I do not see a way of dealing with this in this library at this time. You may want to engage the Next.js team in case you really want to use App Router and maybe this is a scenario they would be interested in supporting.

Please reach out if you have other ideas how to make this work. My best guess right now is to do something like point (5) above. There is possible hope there.

Best of luck!
